### DISCUSSION TITLE
Web3auth for new SDK seems to be generating wallets that no longer show up in torus

### DISCUSSION THREAD

**Discussion by chusla** [Web3Auth/Web3Auth#1249] [Help]
Hello, i am transitioning to the new SDK and noticed something odd.  I am logging in to the same app with my test gmail account and it is 1) generating a new wallet address with nothing else changed apart from SDK version and 2) the new wallet generated doesn't show up when logging into torus with same sign in methodology (same gmail account).  

Does the new SDK generate new wallets for every user using social media and/or email?  If so, is there something special that needs to be done so these are visible in Torus wallet?

**Comment by chusla** [comment]
video of issue i am seeing any input would be much appreciated thank you!  @shahbaz17 
@himanshuchawla009 
https://www.loom.com/share/c6225c444ef146d9b9af490e5160601f

**Comment by chusla** [comment]
seems like official demo app exhibiting same behavior where login with social generates a wallet but then using that same social to log into torus app directly, we are no longer able to see the generated wallet.  Previously on prior SDK you would see whatever wallet's been generated for that social account.

![image](https://user-images.githubusercontent.com/90009543/217055264-8ca8e611-bacf-4947-a183-2e29d605f917.png)
![image](https://user-images.githubusercontent.com/90009543/217055299-d155e427-b66d-4dd9-93ba-e8c15cb974af.png)

**Comment by chusla** [comment]
Here is a video exhibiting the behavior on the w3a demo app.  I log in with social media, click torus it shows one distinct wallet.  if i log out and then log back in with same gmail it shows me a whole other set of wallets and the generated one no longer retrievable.  So question is why are the separate and how can i actually retrieve the one generated by new SDK?  thanks!

https://www.loom.com/share/c6ea348145c14c009db4ae8877afcf6e

**Comment by chusla** [comment]
CW suggested looking at interoperability and turning that on.

double checked and interoperability was turned on for evm chains, i turned on for all other chains.

still getting an issue where a login with gmail account in demo app vs torus app result in completely different sets of wallet addresses.  

another loom with audio discussing the issue i'm seeing.  Is no one else seeing this issue?

https://www.loom.com/share/037cc8b6462746ec9ba4068be345798a

**Reply by ieow** [reply]
> https://www.loom.com/share/037cc8b6462746ec9ba4068be345798a

Select `mainnet` on `Openlogin Network`, you should able to get all the accounts in the `mainnet`

In the video, `testnet` is selected, thus `testnet` accounts is shown.
you can check the `testnet` account here
https://testing.tor.us

**Reply by chusla** [reply]
> > https://www.loom.com/share/037cc8b6462746ec9ba4068be345798a
> 
> Select `mainnet` on `Openlogin Network`, you should able to get all the accounts in the `mainnet`
> 
> In the video, `testnet` is selected, thus `testnet` accounts is shown. you can check the `testnet` account here https://testing.tor.us

thx for response oh interesting @ieow , this is promising, but when i log into testing.tor.us, still seeing my original wallets and not the newly generated one.  is there another way to toggle between "testnet" and "mainnet"?

![image](https://user-images.githubusercontent.com/90009543/218038209-d2a16b4e-6073-499d-a5a9-dc2fd4748721.png)

**Reply by ieow** [reply]
Dapp are only able to access dapp specific wallet. (which is the only one shown) 
where app.tor.us ( login thru app.tor.us) would show all wallets ( dapp that tick interoperability )

login to app.tor.us will show all wallets ( include dapp that allow interoperability ) that run live on mainnet
login to testing.tor.us will do the same for dapp that is testing on testnet

*note: wallet created on testnet are not migrate able to mainnet and vice versa

**Reply by chusla** [reply]
ok i can see that that makes sense, but then how do i get app.torus to display that wallet?  Logging with same gmail to app.torus does not display any wallets generated when logging in with the new sdk for some reason.  thx!

**Reply by ieow** [reply]
I don't get what you mean.
but in order for your dapp wallet to be shown on the app.tor.us
you need to make sure 
- tick interoperability on the dashboard, make sure your project is on mainnet
- on the sdk make sure you configure it to connect to mainnet

**Reply by chusla** [reply]
so in the screenshot below, I logged in via gmail with one2all.test@gmail.com into the web3auth demo app.  it created a wallet starting with 0x178d.   

I separately logged into both app.tor.us and testing.tor.us with the same method (gmail with one2all.test@gmail.com) and was not able to find the wallet 0x178d in the list.  

This occurs even with my own app where i have checked off interoperability (which i do for every client id i generate) and the app is properly configured to be on mainnet.

**Comment by chusla** [comment]
the observed issue in one screenshot.  How can same login methodology in torus app result in two different and discrete sets of wallets?

![image](https://user-images.githubusercontent.com/90009543/217466850-8091168d-b9bf-478c-902d-86ff3d1c0187.png)

**Reply by ieow** [reply]
I speculate that the small window is login on the site demo-app.web3auth.io while the full window is login on app.tor.us

refer this 
https://github.com/orgs/Web3Auth/discussions/1249#discussioncomment-4941205

**Comment by chusla** [comment]
ok even weirder, i ask a question in support and it shows up in both places.  so the account is def the same account.  so why are the sets of wallets completely different?  I'm completely thrown for a loop.  please help!

![image](https://user-images.githubusercontent.com/90009543/217468416-5ea91f50-8351-4a62-97d9-741741a70158.png)

**Comment by ieow** [comment]
In the video, `testnet` is selected, thus `testnet` accounts is shown.
you can check the `testnet` account here
https://testing.tor.us/

Select `mainnet` on `Openlogin Network`, you should able to get all the accounts in the mainnet

**Comment by chusla** [comment]
hmm thanks @ieow this is a promising suggestion but trying it out and still getting same issue.  so switching to "mainnet" i still see the single generated wallet.

Whereas in testing.tor.us still seeing the other original wallets.

![image](https://user-images.githubusercontent.com/90009543/218038830-744c8310-8ced-441c-a278-ed62a786607c.png)
![image](https://user-images.githubusercontent.com/90009543/218038877-4c99ce9c-21b2-4f45-bf8c-9a517c7ed94d.png)

**Comment by chusla** [comment]
@ieow loom with another walk through.  playing around i'm able to get my old wallets if i eliminate all external wallets besides torus-evm.

however if i use "openlogin" to log in this is where i start to get the separate wallet that was generated and the only way i can currently see that within torus is through the pop up window embedded in the demo app.  logging into testing.tor.us or app.tor.us both show my other older wallets.

https://www.loom.com/share/64c8fecb94b9492f985bfe75488389c3

**Comment by chusla** [comment]
ok, messing with url's further.  looks like there is one parameter that is making a big difference and it is "namespace"

if i include "namespace=demo-app.web3auth.io" in app.tor.us url i can access the generated wallet.  but if i do not include it i see my old wallets.  so question now is what is namespace and why is it so important?

https://www.loom.com/share/4d66b48a688b49af912a2207a1f87fe2

![image](https://user-images.githubusercontent.com/90009543/218045050-8baf6956-714d-48e7-b298-3a79c84f8870.png)


![image](https://user-images.githubusercontent.com/90009543/218045081-a84fe9ea-f3cd-46f2-83dd-82f5796f6805.png)

**Reply by chusla** [reply]
any updates on this?  still seeing this issue.  the wallets generated with social seem to be "orphaned" there is no way to retrieve them through app.tor.us

**Reply by ieow** [reply]
For the dapp that host at localhost, *.web3auth.io and *.tor.us, the wallet generated will not show on the app.tor.us
if your dapp is running on localhost, you could use ngrok and whitelist the url to login.
Then the wallet will show at the app.tor.us

**Reply by chusla** [reply]
thanks for the response @ieow is this a new behavior with new SDK?  Because as far as i understand previously wallets weren't silo'ed like this, at least i never encountered an issue where new wallets were being generated and then not accessible/visible via app.tor.us.  

For example, i created a custom dapp which is hosted by a third party (bubble.io) and mapped to my personal domain for instance at demo.one2all.io.  I whitelisted this domain in plug and play for that client id.  But it is still not visible in app.tor.us.  Is this expected behavior?

**Reply by chusla** [reply]
@ieow i think i might have seen you or someone from team log in,  please use this link as it is the one set to latest SDK:

https://demo.one2all.io/version-web3auth-sdk/

Thanks!

it can connect but is generating a new wallet for me relative to prior SDK.

**Reply by chusla** [reply]
or this branch, i just pointed this branch to latest SDK as well:  https://demo.one2all.io/version-test/.  This branch is using testnet client ID whereas prior link "version-web3auth-sdk" is using my mainnet client ID.

So what i'm seeing here is that a new wallet is generated when connecting to https://demo.one2all.io/version-test/ with address:  0x91f6339d2d2d8ebb37a423203abacfbc6a19caab.  this is for gmail login one2all.test@gmail.com

but when i log into app.tor.us or testing.tor.us  that address is not visible or selectable in dropdown.

![image](https://user-images.githubusercontent.com/90009543/220792862-b114d99e-1159-49a0-83e9-089385a3bf55.png)

![image](https://user-images.githubusercontent.com/90009543/220792921-47919fc7-a76a-47ae-b0c7-1069a8263eb5.png)

**Reply by ieow** [reply]
> thanks for the response @ieow is this a new behavior with new SDK? Because as far as i understand previously wallets weren't silo'ed like this, at least i never encountered an issue where new wallets were being generated and then not accessible/visible via app.tor.us.
> 
> For example, i created a custom dapp which is hosted by a third party (bubble.io) and mapped to my personal domain for instance at demo.one2all.io. I whitelisted this domain in plug and play for that client id. But it is still not visible in app.tor.us. Is this expected behavior?

![Screenshot 2023-02-23 at 9 28 50 AM](https://user-images.githubusercontent.com/4881057/220802215-e445faab-6bad-422e-b766-dfea5d593447.png)


It show up on my app.tor.us either for 

https://demo.one2all.io/version-web3auth-sdk/
or
https://demo.one2all.io

**Comment by chusla** [comment]
just connected with my personal gmail account and had same result, a wallet was generated in testnet but was not visible in app.tor.us.  Is it because app,tor.us is only showing mainnet wallets?

testing.tor.us doesn't seem to be pointing at testnet wallets is there a way to force app.tor.us to show testnet wallets?

**Reply by ieow** [reply]
testing.tor.us is not pointing to testnet for now. Will be fixed soon.
Thanks for pointing out

**Reply by ieow** [reply]
testing.tor.us is intended to point to mainnet for testing now.
Please use https://lrc.tor.us/ for testnet

**Reply by chusla** [reply]
Ah that makes more sense now I thought I was going insane!!

I just tried the second site looks like it's not exactly syncing there either maybe the test cluster is not actually testnet not seeing my testnet wallet there either.

Is there timing as to when testing.tor.us will actually point to testnet?  Knowing it will eventually is def comforting but timing would be much better so I can communicate to my clients.  Thanks!

**Comment by chusla** [comment]
there seems to be a precedent for this where certain wallets that are visible in current app.tor.us aren't visible if manually reverted back to like v1.21.0 or earlier via url.

https://app.tor.us/v1.21.0/wallet/home

**Reply by ieow** [reply]
Yes, due to requirement changes, we change the wallet account visibility in app.tor.us

**Comment by chusla** [comment]
Ahhhhh I think mystery solved.  Yes if testing.tor.us pointed we should be
good!  Thank you!

**Comment by chusla** [comment]
Will try thanks!

**Comment by chusla** [comment]
Hi @ieow do you have any timing as to when testing.tor.us will reflect testnet environment wallets?  It's still critical for testing purposes to have full range of functionality including ability to export private key via torus dashboard for example.  Thx!

**Comment by zk-pay** [comment]
Did you figure out how to get the correct account to show on app.tor.us? We have a production deployment with 'cyan' but when users log into app.tor.us, the address is not the same that is used with our modal.

**Reply by lukaskai** [reply]
Have you figured this out?

**Reply by chusla** [reply]
hmm they said at some point testing.tor.us would point to testnet wallet addresses but i haven't seen it yet.

**Reply by lukaskai** [reply]
But I am talking about production setup now, which should work?

**Reply by chusla** [reply]
Production is ok yes.  It seems like a one time thing since they changed their backend environment framework.  Going forward shouldn't change wallets in production from what I understand

**Reply by lukaskai** [reply]
Well at least I try it in production and I still get different wallets and no option to choose my project in the accounts dropdown list :/

**Comment by atifnimran** [comment]
Right now
